Integrations/AWS-S3/integration-AWS-S3.py

```diff
@@ -0,0 +1,368 @@
+    )
+    data = io.BytesIO()
+    client.download_fileobj(args.get('bucket').lower(), args.get('key'), data)
+
+    demisto.results(fileResult(args.get('key'), data.getvalue()))
+
+
+def list_objects_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    data = []
+    response = client.list_objects(Bucket=args.get('bucket'))
+    for key in response['Contents']:
+        data.append({
+            'Key': key['Key'],
+            'Size': convert_size(key['Size']),
+            'LastModified': datetime.strftime(key['LastModified'], '%Y-%m-%dT%H:%M:%S')
+        })
+
+    ec = {'AWS.S3.Buckets(val.BucketName === args.get("bucket")).Objects': data}
+    human_readable = tableToMarkdown('AWS S3 Bucket Objects', data)
+    return_outputs(human_readable, ec)
+
+
+def get_file_path(file_id):
+    filepath_result = demisto.getFilePath(file_id)
+    return filepath_result
+
+
+def upload_file_command(args):
+    client = aws_session(
+        region=args.get('region'),
+        roleArn=args.get('roleArn'),
+        roleSessionName=args.get('roleSessionName'),
+        roleSessionDuration=args.get('roleSessionDuration'),
+    )
+    path = get_file_path(args.get('entryID'))
+
+    with open(path['path'], 'rb') as data:
+        client.upload_fileobj(data, args.get('bucket'), args.get('key'))
+        demisto.results('File {file} was uploaded successfully to {bucket}'.format(
+            file=args.get('key'), bucket=args.get('bucket')))
+
+
+"""COMMAND BLOCK"""
+try:
+    LOG('Command being called is {command}'.format(command=demisto.command()))
+    if demisto.command() == 'test-module':
+        client = aws_session()
+        response = client.list_buckets()
+        if response['ResponseMetadata']['HTTPStatusCode'] == 200:
+            demisto.results('ok')
+
+    elif demisto.command() == 'aws-s3-create-bucket':
+        create_bucket_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-delete-bucket':
+        delete_bucket_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-list-buckets':
+        list_buckets_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-get-bucket-policy':
+        get_bucket_policy_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-put-bucket-policy':
+        put_bucket_policy_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-delete-bucket-policy':
+        delete_bucket_policy_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-download-file':
+        download_file_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-list-bucket-objects':
+        list_objects_command(demisto.args())
+
+    elif demisto.command() == 'aws-s3-upload-file':
+        upload_file_command(demisto.args())
+
+except Exception as e:
```

Comment:
you should use the same handling for region/bad address error that you did in ec2